src/object/srv_obj_migrate.c

```diff
@@ -2151,6 +2227,30 @@ migrate_obj_iter_cb(daos_handle_t ih, d_iov_t *key_iov, d_iov_t *val_iov,
 	if (arg->pool_tls->mpt_fini)
 		return 1;
 
+	if (arg->pool_tls->mpt_del_local_objs) {
+		/* TODO:
+		 *
+		 * This code needs to move. The arg->cont_hdl here is not the
+		 * correct container handle (here it's the client container
+		 * handle, which isn't useful for local deletes).
+		 *
+		 * This will need to either open the container here... or
+		 * move this entire code down underneath somewhere where the
+		 * local server container handle is available (migrate_dkey)
+		 */
+		rc = destroy_existing_obj(arg->pool_tls, oid, arg->cont_hdl);
+		if (rc) {
+			/* An error will only be thrown here for an important
+			 * reason - no error will be generated if the object
+			 * simply can't be deleted. Any error returned here
+			 * should halt this object migration.
+			 */
+			D_ERROR("destroy_existing_obj failed: "DF_RC"\n",
+				DP_RC(rc));
+			return rc;
+		}
+	}
+
```

Comment:
@wangdi1 This is the code I could use your advice on where to put. 

Either leave it here, and open the container here so that object deletes are possible, or move it under migrate_dkey() where ds_cont_child_open_create() is called.

Reply:
Hmm, this is system xstream. I do not think you can call vos_obj_delete() on this xstream. So you probably have to delete the object on migrate_dkey().  Plus the target, where migrate_dkey() is executed, is the one which is being reintegrated, i.e. only the object on that target can be deleted, and you can not delete object on other targets anyway.

Reply:
hmm, you actually can delete the object here too, but you need do vos_obj_delete on the target where the reintegrate shard is.

Reply:
Thanks, totally new approach in the next patch that puts this in the right place.